internal/util/msg.go

```diff
@@ -65,3 +66,27 @@ func WriteErr(n int, err error) {
 		log.Errorf("write: %s", err)
 	}
 }
+
+func ReadBytes(c net.Conn, b []byte, n int) (int, error) {
+	l := 0
+	for l < n {
+		tmp, err := c.Read(b[l:])
+		if err != nil {
+			return l, err
```

Comment:
Should be `l + tmp`?

Reply:
The data in `b[l: l + tmp]` may be unreliable because the reading errror.